Build Issue - can't build with networking support on Alpine

## Build Process
Following the [instructions](https://github.com/jaagr/polybar/wiki/Compiling) as precisely as possible.

*Describe how you build polybar, list the exact commands you are using:*
I used the exact commands from that document:
```
git clone --recursive https://github.com/jaagr/polybar
mkdir polybar/build
cd polybar/build
cmake ..
sudo make install
```
**If you build polybar directly from this repository:**

* Output of `git describe --tags`:
3.2.1

**If you use some other way (like the AUR):**
nope
*List exactly where you are building from and which version you are building.*


## Build Log
*Post everything that is output to the terminal while building polybar below. This HAS to include the output of the `cmake` and `make` commands, if you used them.*
```
cmake output...
-- Trying to enable ccache
-- Using compiler cache /usr/bin/ccache
-- The C compiler identification is GNU 6.4.0
-- The CXX compiler identification is GNU 6.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type specified; using RelWithDebInfo
-- Using supported compiler GNU-6.4.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.5.3") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.15.12
-- Checking for module 'alsa'
--   Found alsa, version 1.1.6
-- Checking for module 'libnl-genl-3.0'
--   Found libnl-genl-3.0, version 3.4.0
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.13
-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.13
-- Checking for module 'xcb-randr>=1.12'
--   Found xcb-randr, version 1.13
--  Build:
--    Type: RelWithDebInfo
--    CC: /usr/bin/cc  -O2 -g -DNDEBUG
--    CXX: /usr/bin/c++  -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors -O2 -g -DNDEBUG
--    LD: /usr/bin/ld  
--  Targets:
-- [X]   polybar-msg
-- [ ]   testsuite
--  Module support:
-- [X]   alsa (1.1.6)
-- [ ]   curl
-- [X]   i3
-- [ ]   mpd
-- [X]   network (3.4.0)
-- [ ]   pulseaudio
-- [X]   xkeyboard (1.13)
--  X extensions:
-- [X]   xcb-randr (1.13)
-- [X]   xcb-randr (monitor support) (1.13)
-- [ ]   xcb-render
-- [ ]   xcb-damage
-- [ ]   xcb-sync
-- [ ]   xcb-composite
-- [X]   xcb-xkb (1.13)
-- [ ]   xcb-xrm
-- [ ]   xcb-cursor
--  Log options:
-- [X]   Trace logging
-- Found font: fixed:pixelsize=10
-- Found font: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.13
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.15", minimum required is "2.7") 
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/libxcb.so  
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/libxcb-icccm.so  
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/libxcb-ewmh.so  
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/libxcb-util.so  
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/libxcb-image.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so  
-- Searching for xcbgen with python2
-- Searching for xcbgen with python2.7
-- Searching for xcbgen with python3
-- Found xcbgen in /usr/lib/python3.6/site-packages
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/libxcb-randr.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/lib/libxcb-randr.so  
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/libxcb-xkb.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/lib/libxcb-randr.so;/usr/lib/libxcb-xkb.so  
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp>=1.7.7'
--   Package 'jsoncpp', required by 'virtual:world', not found
-- Could not find jsoncpp>=1.7.7 -- building static lib...
-- JsonCpp Version: 1.7.7
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.15", minimum required is "2.6") 
-- Configuring done
-- Generating done

make output...
[  4%] Built target xpp
[  9%] Built target jsoncpp_lib_static
[ 12%] Built target i3ipc++
[ 13%] Building CXX object bin/CMakeFiles/polybar.dir/adapters/net.cpp.o
/home/david/Projects/Software/polybar/src/adapters/net.cpp: In member function 'void polybar::v3_2_1::net::network::check_tuntap()':
/home/david/Projects/Software/polybar/src/adapters/net.cpp:197:41: error: 'caddr_t' does not name a type
     request.ifr_data = reinterpret_cast<caddr_t>(&driver);
                                         ^~~~~~~
/home/david/Projects/Software/polybar/src/adapters/net.cpp: In member function 'virtual bool polybar::v3_2_1::net::wired_network::query(bool)':
/home/david/Projects/Software/polybar/src/adapters/net.cpp:261:41: error: 'caddr_t' does not name a type
     request.ifr_data = reinterpret_cast<caddr_t>(&data);
                                         ^~~~~~~
/home/david/Projects/Software/polybar/src/adapters/net.cpp: In member function 'virtual bool polybar::v3_2_1::net::wired_network::connected() const':
/home/david/Projects/Software/polybar/src/adapters/net.cpp:286:41: error: 'caddr_t' does not name a type
     request.ifr_data = reinterpret_cast<caddr_t>(&data);
                                         ^~~~~~~
At global scope:
cc1plus: error: unrecognized command line option '-Wno-noexcept-type' [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [bin/CMakeFiles/polybar.dir/build.make:89: bin/CMakeFiles/polybar.dir/adapters/net.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:822: bin/CMakeFiles/polybar.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

## Environment:
* Distro (with Version):
Alpine Linux 3.8

## Additional context
*Add any other context that you think is necessary about the problem here.*
When searching for general info about 'caddr-t' I found references to that type being rejected from becoming a posix standard and should be replaced with 'void *'. No idea if this idea has merit.

Wondering if I should just try the replacement and re-build, but the readme says to report any build issues, so here I am.

--- a/src/adapters/net.cpp
+++ b/src/adapters/net.cpp
@@ -194,7 +194,7 @@ namespace net {
      */
     strncpy(request.ifr_name, m_interface.c_str(), IFNAMSIZ - 1);
 
-    request.ifr_data = reinterpret_cast<caddr_t>(&driver);
+    request.ifr_data = reinterpret_cast<char*>(&driver);
 
     if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
       return;
@@ -258,7 +258,7 @@ namespace net {
     memset(&request, 0, sizeof(request));
     strncpy(request.ifr_name, m_interface.c_str(), IFNAMSIZ - 1);
     data.cmd = ETHTOOL_GSET;
-    request.ifr_data = reinterpret_cast<caddr_t>(&data);
+    request.ifr_data = reinterpret_cast<char*>(&data);
 
     if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
       return false;
@@ -283,7 +283,7 @@ namespace net {
     memset(&request, 0, sizeof(request));
     strncpy(request.ifr_name, m_interface.c_str(), IFNAMSIZ - 1);
     data.cmd = ETHTOOL_GLINK;
-    request.ifr_data = reinterpret_cast<caddr_t>(&data);
+    request.ifr_data = reinterpret_cast<char*>(&data);
 
     if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
       return false;

